### PR TITLE
Task/backend48 setup rtmp server

### PR DESCRIPTION
**Pull request by shirokuronemuri**

### CODE REVIEW

File: .docker/docker-compose.dev.yml [hunk 1]
```diff
@@ -1,5 +1,17 @@
...
+  rtmp:
+    image: tiangolo/nginx-rtmp
+    ports:
+      - "1935:1935"
+    container_name: rtmp_server
+    depends_on:
+      - backend
+    networks:
+      - backend_network
+      - rtmp_network
```
Comment (VitaliiBedletskyi) [inline .docker/docker-compose.dev.yml]: Get rig of rtmp_network...

File: .docker/docker-compose.local.yml [hunk 2]
```diff
@@ -1,5 +1,17 @@
...
+  rtmp:
+    image: tiangolo/nginx-rtmp
+    ports:
+      - "1935:1935"
+    container_name: rtmp_server
+    depends_on:
+      - backend
+    networks:
+      - backend_network
+      - rtmp_network
```
Comment (VitaliiBedletskyi) [inline .docker/docker-compose.local.yml]: unnecessary network

File: backend/core/src/core/channel/application/channel-service.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    return this.amqpChannel.sendToQueue({
+      queue: AmqpQueue.STREAM_TRANSCODER,
+      content: Buffer.from(
+        JSON.stringify({
+          streamingKey,
+          isLive: false,
+        }),
+      ),
+    });
+  }
```
Comment (VitaliiBedletskyi) [inline backend/core/src/core/channel/application/channel-service.ts:62]: Please sync with @WanderningMaster  about if is it really necessary

File: backend/core/src/primary-adapters/rest/channel/channel-controller.ts [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+   *          description: Streaming token is wrong or the channel doesn't have any pending streams
+   */
+  @httpPost(ChannelApiPath.LIVE)
+  public async goLive(@requestBody() rtmpLiveRequestDto: RtmpLiveRequestDto): Promise<void> {
+    const streamData = await this.channelService.checkStreamingKey(rtmpLiveRequestDto.name);
+    if (streamData === null) {
+      throw new Forbidden('Invalid streaming key or no video created to stream on');
+    }
+    this.channelService.notifyTranscoderAboutStreamStart(streamData);
+    return;
```
Comment (VitaliiBedletskyi) [inline backend/core/src/primary-adapters/rest/channel/channel-controller.ts]: return statement is unnecessary

File: backend/core/src/primary-adapters/rest/channel/channel-controller.ts [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+   *          schema:
+   *            $ref: '#/components/schemas/RtmpLiveRequestDto'
+   *      responses:
+   *        '200':
+   *          description: new streaming key is given
+   */
+  @httpPost(ChannelApiPath.LIVE_DONE)
+  public async prepareStreamEnd(@requestBody() rtmpLiveRequestDto: RtmpLiveRequestDto): Promise<void> {
+    this.channelService.notifyTranscoderAboutStreamEnd(rtmpLiveRequestDto.name);
+    return;
```
Comment (VitaliiBedletskyi) [inline backend/core/src/primary-adapters/rest/channel/channel-controller.ts]: Also this one

File: backend/core/src/secondary-adapters/postgres/channel/channel-repository-adapter.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        status: StreamingStatus.PENDING,
+      },
+    });
+    if (!pendingStream) {
+      return null;
+    }
+    return {
+      videoId: pendingStream.id,
+      streamingKey: key,
+    };
```
Comment (VitaliiBedletskyi) [inline backend/core/src/secondary-adapters/postgres/channel/channel-repository-adapter.ts]: The adapters shouldn't do any business logic, they only have getters and setters for infrastructure like databases, message brokers, etc. All business logic should be in services. Please do refactor all methods

**Review by VitaliiBedletskyi — CHANGES_REQUESTED** [review]

File: backend/core/src/core/channel/application/channel-service.ts (hunk 3 above)
Comment (shirokuronemuri) [reply backend/core/src/core/channel/application/channel-service.ts:62]: yeah, already done it

**Review by VitaliiBedletskyi — APPROVED** [review]

**Comment by shirokuronemuri** [comment]
[Setup RTMP server](https://trello.com/c/9wty1l6U/48-setup-rtmp-server)

**Pull request merged**